wicket-core/src/main/java/org/apache/wicket/page/PartialPageUpdate.java

```diff
@@ -657,34 +747,11 @@ public void render(HeaderItem item)
 				item = ((IWrappedHeaderItem) item).getWrapped();
 			}
 
-			if (item instanceof OnLoadHeaderItem)
-			{
-				if (!wasItemRendered(item))
-				{
-					PartialPageUpdate.this.appendJavaScript(((OnLoadHeaderItem) item).getJavaScript());
-					markItemRendered(item);
-				}
-			}
-			else if (item instanceof OnEventHeaderItem)
+			if (item instanceof OnLoadHeaderItem || item instanceof OnEventHeaderItem || item instanceof OnDomReadyHeaderItem)
```

Comment:
Why OnDomReadyHeaderItem is not handled with `PartialPageUpdate.this.domReadyJavaScripts.add()` anymore ?